wp_api/src/login.rs

```diff
@@ -332,6 +363,39 @@ mod tests {
             .is_empty());
     }
 
+    #[rstest]
+    #[case("api-details/invalid-json-01.json")]
+    #[case("api-details/invalid-json-02.json")]
+    fn test_valid_json_documents(#[case] input: &str) {
+        let json = test_json(input).expect("Failed to read test resource");
+
+        let result = WpApiDetails::try_from(json);
+
+        if result.is_err() {
+            // Logging the error should give us a line number or missing field name to figure out the problem
+            println!(
+                "Failed to parse json as `WpApiDetails`: {}",
+                result.as_ref().err().unwrap()
+            );
+        }
+
+        assert!(result.is_ok());
+    }
+
+    fn test_json(input: &str) -> Result<Vec<u8>, std::io::Error> {
+        let mut file_path = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        file_path.push("tests");
```

Comment:
The path `./tests` have a special meaning for cargo. Could we rename the folder to `test-data`?

I am a little reluctant to do IO as part of unit tests, but I think it's fine to leave it as is for now.

Reply:
Addressed in https://github.com/Automattic/wordpress-rs/pull/598/commits/a3e84220c137e41bb5ae6d38c18ad0f5c1fda868

I think disk-based IO is generally ok – I'd rather keep the tests readable for parsing larger JSON docs